test/e2e/operator.go

```diff
@@ -379,36 +387,125 @@ var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
 })
 
 var _ = Describe("ARO Operator - MUO Deployment", func() {
-	ctx := context.Background()
+	When("we have a fresh ARO cluster", func() {
+		It("must deploy MUO by default with FIPS crypto mandated", func() {
+			hasMUOFIPSCryptoMandatedFn := hasMUOFIPSCryptoMandatedHigherOrderFn(context.Background())
 
-	It("must be deployed by default with FIPS crypto mandated", func() {
-		muoIsDeployed := func() (bool, error) {
-			By("getting MUO pods")
-			pods, err := clients.Kubernetes.CoreV1().Pods("openshift-managed-upgrade-operator").List(ctx, metav1.ListOptions{
-				LabelSelector: "name=managed-upgrade-operator",
-			})
-			if err != nil {
-				return false, err
-			}
-			if len(pods.Items) != 1 {
-				return false, fmt.Errorf("%d managed-upgrade-operator pods found", len(pods.Items))
-			}
+			err := wait.PollImmediate(interval, duration, hasMUOFIPSCryptoMandatedFn)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
 
-			By("getting logs from MUO")
-			b, err := clients.Kubernetes.CoreV1().Pods("openshift-managed-upgrade-operator").GetLogs(pods.Items[0].Name, &corev1.PodLogOptions{}).DoRaw(ctx)
-			if err != nil {
-				return false, err
-			}
+	When("MUO is deleted on the ARO cluster", func() {
+		It("should reconcile the MUO deployment", func() {
+			ctx := context.Background()
+			doesMUODeploymentExistFn := doesMUODeploymentExistHigherOrderFn(ctx)
```

Comment:
Could we just inline the function below? Not sure if it is easier to read with that name

Reply:
Declaring a function inside another function is something I don't really like, neither in production nor in tests, as it pollutes the test's assertions with business logic you are not really interested to see at first sight. It gets even worse when the inlined function is big (> 10 lines).

When I read a test, I want to see something like [arrange, act, assert] in a straightforward way without worrying about how the act part is performed (if I want to see the details I jump to the function to see those details). 

That's why I created the High Order Functions to simplify the content of the tests. Ideally, I would directly define the functions outside the tests, but I need the High Order Functions (as a wrapper) because I need to access an outer variable (ctx) that is not passed in the wait.ConditionFunc type definition but that I need in the function's body.

After your comment, I have simplified the code and know it reads a bit better. 
😃